apiml/src/test/java/org/zowe/apiml/ZaasSchemeTransformApiTest.java

```diff
@@ -46,294 +41,321 @@ class ZaasSchemeTransformApiTest {
     private AuthSourceService authSourceService;
     private PassTicketService passTicketService;
     private ZaasSchemeTransformApi transformApi;
+    TokenCreationService tokenCreationService;
+    ZosmfService zosmfService;
+    RequestCredentials credentials;
+    AuthSource authSource;
     private final MessageService messageService = new YamlMessageService("/apiml-log-messages.yml");
 
+    private final String INVALID_AUTH_MSG = "ZWEAO402E The request has not been applied because it lacks valid authentication credentials.";
+    private final String MISSING_AUTH_MSG = "ZWEAG160E No authentication provided in the request";
+
     @BeforeEach
     void setUp() {
         authSourceService = mock(AuthSourceService.class);
+        when(authSourceService.isValid(any())).thenReturn(true);
+        tokenCreationService = mock(TokenCreationService.class);
+        zosmfService = mock(ZosmfService.class);
         passTicketService = mock(PassTicketService.class);
-        ZosmfService zosmfService = mock(ZosmfService.class);
-        TokenCreationService tokenCreationService = mock(TokenCreationService.class);
-
+        credentials = mockCredentials();
+        authSource = mock(AuthSource.class);
         transformApi = new ZaasSchemeTransformApi(
             authSourceService,
             passTicketService,
-            zosmfService,
+            mock(ZosmfService.class),
             tokenCreationService,
             messageService
         );
     }
 
+
     @Nested
-    class GivenPassticket {
-        @Test
-        void thenReturnsExpectedTicket() throws PassTicketException {
-            RequestCredentials credentials = mockCredentials();
+    class GivenPassticketScheme {
+        @Nested
+        class GivenValidAuth {
 
-            AuthSource authSource = mock(AuthSource.class);
-            AuthSource.Parsed parsed = mock(AuthSource.Parsed.class);
-            when(parsed.getUserId()).thenReturn("USER1");
+            @BeforeEach
+            public void setup() {
 
+                var parsed = mock(AuthSource.Parsed.class);
```

Comment:
It looks like this block is the same for each nested class,  so it could be maybe put on the higher level, but...

Reply:
it is, but then it colides with tests that require no auth but should be under different group, I don't think it is so crucial to not repeat code in test setup